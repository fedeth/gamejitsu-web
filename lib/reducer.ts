const initialState = {}
type State = typeof initialState 

const reducer = (state = {}, action): State => {
  switch (action.type) {
    default:
      return state
  }
}

export default reducer
