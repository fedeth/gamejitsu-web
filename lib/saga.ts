import { all, AllEffect, ForkEffect } from 'redux-saga/effects'

export default function*(): IterableIterator<AllEffect<IterableIterator<ForkEffect>>> {
  yield all([])
}
