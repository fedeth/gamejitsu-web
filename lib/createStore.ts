import { createStore as reduxCreateStore, applyMiddleware, Store as ReduxStore } from 'redux'
import createSagaMiddleware, { Task } from 'redux-saga'
import reducer from './reducer'
import saga from './saga'

export interface State {}

export interface Store extends ReduxStore<State> {
  sagaTask?: Task
}

const createStore = (initialState: State): Store => {
  const sagaMiddleware = createSagaMiddleware()
  const store: Store = reduxCreateStore(reducer, initialState, applyMiddleware(sagaMiddleware))
  store.sagaTask = sagaMiddleware.run(saga)
  return store
}

export default createStore
