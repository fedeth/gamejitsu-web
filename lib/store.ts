import { createStore, applyMiddleware, Store } from 'redux'
import createSagaMiddleware, { Task } from 'redux-saga'
import reducer from './reducer'
import saga from './saga'

const makeStore = (initialState, options): StoreWithSaga => {
  const sagaMiddleware = createSagaMiddleware()
  const store: StoreWithSaga = createStore(reducer, initialState, applyMiddleware(sagaMiddleware))
  store.sagaTask = sagaMiddleware.run(saga)
  return store
}

export interface StoreWithSaga extends Store {
  sagaTask?: Task
}

export default makeStore
