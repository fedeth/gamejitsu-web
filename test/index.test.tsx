import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { theme } from '../lib/theme'
import Home from '../pages/index'

test('Home Test', () => {
  const home = mount(
    <ThemeProvider theme={theme}>
      <Home />
    </ThemeProvider>
  )
  expect(home).toMatchSnapshot()
})
