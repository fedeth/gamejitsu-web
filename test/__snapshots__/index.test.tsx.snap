// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#1D0D0D",
      "primaryColor": "#ADD90D",
      "secondaryColor": "#488C03",
      "textColor": "#F1F2E9",
      "textFont": "'Exo 2',sans-serif",
    }
  }
>
  <Home>
    <div>
      <div>
        Welcome to Next.js!
      </div>
      <h1>
        Gamejitsu
      </h1>
      <Button
        text="Send Replay"
      >
        <styled.button>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "hGTcSD",
                  "rules": Array [
                    "
  background: linear-gradient(
    to bottom,
    ",
                    [Function],
                    ",
    ",
                    [Function],
                    "
  );
  padding: 12px 30px;
  font-size: 25px;
  outline: 0;
  border: 0;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <button
              className="sc-bdVaJa hGTcSD"
            >
              Send Replay
            </button>
          </StyledComponent>
        </styled.button>
      </Button>
    </div>
  </Home>
</ThemeProvider>
`;
